# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1550611901 0
#      Tue Feb 19 21:31:41 2019 +0000
# Branch stable
# Node ID 3d0f687e4a21795df7f9ffc189fa4214d74cd8bc
# Parent  b812ed975ea8822fb21b57d58cdf4b4a36759b7f
avoid crash if help engine setup fails (bug #55736)

* documentation.cc (documentation::documentation): Don't return early
if help engine setup fails.  Do disconnect existing help engine
signals, delete help engine object, and set m_help_engine to nullptr.

diff --git a/libgui/src/documentation.cc b/libgui/src/documentation.cc
--- a/libgui/src/documentation.cc
+++ b/libgui/src/documentation.cc
@@ -107,20 +107,21 @@ namespace octave
     if (! m_help_engine->setupData())
       {
 #ifdef ENABLE_DOCS
         QMessageBox::warning (this, tr ("Octave Documentation"),
                               tr ("Could not setup the data required for the\n"
                                   "documentation viewer. Only help texts in\n"
                                   "the Command Window will be available."));
 #endif
-        if (m_help_engine)
-          delete m_help_engine;
-        m_help_engine = 0;
-        return;
+
+        disconnect (m_help_engine, 0, 0, 0);
+
+        delete m_help_engine;
+        m_help_engine = nullptr;
       }
 
     // The browser
     QWidget *browser_find = new QWidget (this);
     m_doc_browser = new documentation_browser (m_help_engine, browser_find);
     connect (m_doc_browser, SIGNAL (cursorPositionChanged (void)),
              this, SLOT(handle_cursor_position_change (void)));
 
